services/director-v2/src/simcore_service_director_v2/modules/osparc_variables/substitutions.py

```diff
@@ -130,11 +130,9 @@ def create(cls, app: FastAPI):
 
         table.register_from_handler("OSPARC_VARIABLE_USER_EMAIL")(request_user_email)
         table.register_from_handler("OSPARC_VARIABLE_USER_ROLE")(request_user_role)
-        table.register_from_handler("OSPARC_VARIABLE_API_KEY")(
-            get_or_create_user_api_key
-        )
+        table.register_from_handler("OSPARC_VARIABLE_API_KEY")(create_user_api_key)
         table.register_from_handler("OSPARC_VARIABLE_API_SECRET")(
-            get_or_create_user_api_secret
+            create_user_api_secret
```

Comment:
Q: are you sure about this?
Please start a service which requires API_KEY and API_SECRET and see if the provided credentials are working.

My doubt is: will the KEY and SECRET be part of the same pair or different ones.

Reply:
Difference ones, but we strongly rely on garbage collection of unused API Keys (#5875)